src/client/tile.h

```diff
@@ -286,5 +296,12 @@ struct TileSpec
 	std::vector<FrameSpec> frames;
 
 	u8 rotation;
+	//! If true, the tile has its own color.
+	bool has_color;
```

Comment:
just a thought: you could make hasColor a method that checks whether the color is non-transparent, this would save you the bool.
```c++
bool TileSpec::hasColor() const {
    return color.getAlpha() > 0;
}
```